flatpak/org.heroic.yml

```
@@ -0,0 +1,65 @@
+app-id: org.hgl.heroic
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node16
+command: run.sh
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --allow=multiarch
+  - --allow=devel
+  - --share=network
+  - --filesystem=host
+  - --talk-name=org.freedesktop.Notifications
+  - --persist=.wine
+  - --filesystem=xdg-data/Steam:ro
+  - --filesystem=~/.var/app/com.valvesoftware.Steam:ro
+build-options:
+  append-path: /usr/lib/sdk/node16/bin
+  cflags: -O2 -g
+  cxxflags: -O2 -g
+  env:
+    NPM_CONFIG_LOGLEVEL: info
+modules:
+  - name: heroic
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/heroic/flatpak-node/cache
+        npm_config_cache: /run/build/heroic/flatpak-node/node-cache
+        npm_config_nodedir: /usr/lib/sdk/node16
+        npm_config_offline: 'true'
+    subdir: main
+    sources:
+      #- type: archive
+      #  url: https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/releases/download/v2.0.0/heroic-2.0.0.tar.xz
+      #  sha256: 5aa8112359ce892ae3e045a56ec90e8fe7a588139ae58ee52284586afac4c021
+      - type: dir
+        path: ..
+        dest: main
+      - generated-sources.json
+      # Wrapper to launch the app
+      - type: script
+        dest-filename: run.sh
+        commands:
+          - export TMPDIR=$XDG_RUNTIME_DIR/app/$FLATPAK_ID
+          - zypak-wrapper.sh /app/main/heroic "$@"
+    build-commands:
+      # Install npm dependencies
+      - HOME=$PWD yarn config --offline set yarn-offline-mirror $FLATPAK_BUILDER_BUILDDIR/flatpak-node/yarn-mirror
+      - yarn --offline
+      # Build the app
+      - |
+        . ../flatpak-node/electron-builder-arch-args.sh
+        yarn dist-flatpak
```

Since you don't pass `$ELECTRON_BUILDER_ARCH_ARGS` to `yarn`, you can remove the bach source thing
```suggestion
      - yarn dist-flatpak
```
Or, if you want the args, actually pass them to yarn
```suggestion
      - |
        . ../flatpak-node/electron-builder-arch-args.sh
        yarn dist-flatpak -- $ELECTRON_BUILDER_ARCH_ARGS
```

---

What about the idea to use the AppImage instead of budding it with Yarn?

---

You can repackage a binary package, sure, but it's safe only as long as the app doesn't have native node modules (otherwise no ABI compatibility guarantees).